data/reusables/actions/jobs/section-using-environments-for-jobs.md

```diff
@@ -27,3 +27,13 @@ environment:
   url: ${{ steps.step_id.outputs.url_output }}
 ```
 {% endraw %}
+
+The `name` can also be an expression but the available contexts are not currently documented.
```

Comment:
> but the available contexts are not currently documented.

I've asked the Actions team about this so that we can provide the details of contexts.

Reply:
Contexts for `name`:
"github","inputs","vars","needs","strategy","matrix"

See: https://github.com/actions/languageservices/blob/main/workflow-parser/src/workflow-v1.0.json#L1965-L1973

---

For precision, we should also update the contexts for `url` to give the actual list of contexts, rather than say "except for the secrets context".

"github","inputs","vars","needs","strategy","matrix","steps","job","runner","env"

See: https://github.com/actions/languageservices/blob/main/workflow-parser/src/workflow-v1.0.json#L2456-L2467

---

`jobs.<job_id>.concurrency` has the same set of contexts as `name` (https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idconcurrency) so we should update this also - replacing "except for the secrets context".

See: https://github.com/actions/languageservices/blob/main/workflow-parser/src/workflow-v1.0.json#L1903-L1913